src/plugins/vis_default_editor/public/components/controls/number_interval.tsx

```diff
@@ -55,21 +83,32 @@ function NumberIntervalParamEditor({
   setTouched,
   setValidity,
   setValue,
-}: AggParamEditorProps<number | undefined>) {
+}: AggParamEditorProps<string | undefined>) {
+  const [autoChecked, setAutoChecked] = useState(isAutoInterval(value));
```

Comment:
Please, don't use internal state for the editor params, it leads to next problems:
- change `Use auto interval` to `false`;
- click on `Discard` changes;
- the editor state is wrong

![histogram_gap](https://user-images.githubusercontent.com/31325372/91727420-5c13f180-ebaa-11ea-8139-5ca3142694ae.gif)

Reply:
Hm... interesting. I fixed that for my simple case by removing `setState` hook, but for me it looks like an issue in editor. Probably we should add a description somewhere that one of `core React features` not working fine for that case. Or fix it somehow. 
Anyway thank you to showing me that scenario

Reply:
There is nothing wrong with React here. Initializing a state value happens only at the first render.
So, when you discard the editor changes - the `value` comes from outside of the component state and it doesn't update the internal component state.

Reply:
I understand, but we are talking about `Discard` button. Probably we should re-render the whole container.
I just wanted to say that setState one of the base features and it's too difficult to understand that for some reason we cannot use it in our Editor